var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -69,23 +77,53 @@ class Mesa(AutotoolsPackage):
 
     # Variant dependencies
     depends_on('llvm@6:', when='+llvm')
+
     depends_on('libx11',  when='+glx')
-    depends_on('libxcb',  when='+glx')
+    depends_on('libxcb@1.9.3:', when='+glx')
     depends_on('libxext', when='+glx')
+
+    depends_on('libglvnd', when='+glvnd')
+
+    depends_on('libxshmfence@1.1:', when='+glvnd')
+
+    plus_dri = ('+egl', '+glvnd')
+    multi_depends('libdrm', when=plus_dri)
+    multi_depends('dri2proto', when=plus_dri)
+
+    depends_on('libxdamage@1.1:', when='+glvnd')
```

Comment:
I'm not entirely sure which of these are for glvnd and which are for dri so you may want to double check. (it might be correct as is, i'm not certain though).

Reply:
dependencies tested using the following:

```
spack configure mesa+glvnd+glx # with glvnd
spack configure mesa+egl+glx # with egl
spack configure mesa+glvnd+egl+glx # with both
spack configure mesa~glvnd~egl+glx # with neither
```